File: jenkins-common/src/main/java/com/app/jenkins_common/lib/IntegerPreferenceType.java
package com.app.jenkins_common.lib;


import android.content.SharedPreferences;

public final class IntegerPreferenceType implements SharedPreferenceType<Integer>{
  private final SharedPreferences sharedPreferences;
  private final int defaultValue = -1;

  public IntegerPreferenceType(SharedPreferences sharedPreferences) {
    this.sharedPreferences = sharedPreferences;
  }

  @Override
  public boolean notNullPref() {
    return sharedPreferences != null;
  }

  @Override
  public Integer get(String key) {
    if (notNullPref()){
      return sharedPreferences.getInt(key,defaultValue);
    } else {
      return defaultValue;
    }
  }

  @Override
  public boolean isSet(String key) {
    return sharedPreferences.contains(key);
  }

  @Override
  public void set(String key, Integer value) {
    if (notNullPref()) {
      final SharedPreferences.Editor editor = sharedPreferences.edit();
      editor.putInt(key,value);
      editor.apply();
    }

  }

  @Override
  public void delete(String key) {
    if (notNullPref()){
      sharedPreferences.edit().remove(key).commit();
    }
  }
}

File: jenkins-common/src/main/java/com/app/jenkins_common/lib/IntegerPreferenceType.kt
package com.app.jenkins_common.lib


import android.content.SharedPreferences

class IntegerPreferenceType(private val sharedPreferences: SharedPreferences?) : SharedPreferenceType<Int> {
    private val defaultValue = -1

    override fun notNullPref(): Boolean {
        return sharedPreferences != null
    }

    override fun get(key: String): Int {
        return if (notNullPref()) {
            sharedPreferences!!.getInt(key, defaultValue)
        } else {
            defaultValue
        }
    }

    override fun isSet(key: String): Boolean {
        return sharedPreferences!!.contains(key)
    }

    override fun set(key: String, value: Int) {
        if (notNullPref()) {
            val editor = sharedPreferences!!.edit()
            editor.putInt(key, value!!)
            editor.apply()
        }
    }

    override fun delete(key: String) {
        if (notNullPref()) {
            sharedPreferences!!.edit().remove(key).commit()
        }
    }
}
